payments-ui-core/src/main/java/com/stripe/android/ui/core/FormUI.kt

```diff
@@ -74,82 +75,161 @@ fun FormUI(
             controller = element.controller,
-            modifier = Modifier.padding(vertical = 4.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 4.dp,
+            ),
         )
         is AfterpayClearpayHeaderElement -> AfterpayClearpayElementUI(
             enabled = enabled,
             element = element,
-            modifier = Modifier.padding(4.dp, 8.dp, 4.dp, 4.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                start = 4.dp,
+                end = 4.dp,
+                top = 8.dp,
+                bottom = 4.dp,
+            ),
         )
         is AuBecsDebitMandateTextElement -> AuBecsDebitMandateElementUI(
             element = element,
-            modifier = Modifier.padding(vertical = 8.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is AffirmHeaderElement -> AffirmElementUI(
-            modifier = Modifier.padding(vertical = 8.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is MandateTextElement -> MandateTextUI(
             element = element,
-            modifier = Modifier.padding(top = element.topPadding, bottom = 8.dp)
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                top = element.topPadding,
+                bottom = 8.dp,
+            ),
         )
         is CardDetailsSectionElement -> CardDetailsSectionElementUI(
-            enabled,
-            element.controller,
-            hiddenIdentifiers,
-            lastTextFieldIdentifier
+            enabled = enabled,
+            controller = element.controller,
+            hiddenIdentifiers = hiddenIdentifiers,
+            lastTextFieldIdentifier = lastTextFieldIdentifier,
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is BsbElement -> BsbElementUI(
             enabled = enabled,
             element = element,
             lastTextFieldIdentifier = lastTextFieldIdentifier,
-            modifier = Modifier.padding(vertical = 8.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is OTPElement -> OTPElementUI(enabled, element)
         is RenderableFormElement -> element.ComposeUI(enabled)
-        is EmptyFormElement -> {}
     }
 }
+
+private fun Modifier.addFormVerticalPadding(
```

Comment:
Nit: Could be just `Modifier.formVerticalPadding()` to be more in line with naming conventions.